stdlib/Random/src/XoshiroSimd.jl

```diff
@@ -222,7 +223,7 @@ end
         res = _plus(_rotl23(_plus(s0,s3)),s0)
         resLoc = _and(res, 0x0101010101010101)
         ref = Ref(resLoc)
-        ccall(:memcpy, Ptr{Cvoid}, (Ptr{UInt8}, Ptr{UInt64}, Csize_t), dst+i, ref, len-i)
+        GC.@preserve ref memcpy(dst+i, Base.unsafe_convert(Ptr{Cvoid}, ref), len-i)
```

Comment:
Why the explicit `@preserve` and `unsafe_convert`?

There is a `cconvert`/`unsafe_convert` chaing defined for `Ptr{Cvoid}`/`Ref{T}`

`Base.unsafe_convert(Ptr{Cvoid}, Base.cconvert(Ptr{Cvoid}, Ref{Int}()))`

Reply:
the `mem*` functions only accept pointers so there's the explicit conversion to a pointer there. I figured we wanted those methods to be consistent with the other `Libc` memory functions that typically only accept pointers.